tools/clang/unittests/HLSL/ExecutionTest.cpp

```diff
@@ -9740,99 +9749,141 @@ TEST_F(ExecutionTest, DynamicResourcesUniformIndexingTest) {
-    }
+      // Update shader target in xml.
+      for (st::ShaderOpShader &S : pShaderOp->Shaders){
+        if (compilerOptions != ""){
+          S.Arguments = pShaderOp->GetString(compilerOptions.c_str());
+        }        
+        // Set the target correctly. Setting here permanently overwrites
+        // the Target string even in future iterations.
+        if (sm==D3D_SHADER_MODEL_6_0){
+          std::string Target(S.Target);
+          Target[Target.length() - 1] = '0';
+          S.Target = pShaderOp->GetString(Target.c_str());
+        }
+        else if (sm==D3D_SHADER_MODEL_6_6){
+          std::string Target(S.Target);
+          Target[Target.length() - 1] = '6';
+          S.Target = pShaderOp->GetString(Target.c_str());
+        }
+      }
 
-    // Test Vertex + Pixel shader
-    {
-      pShaderOp->CS = nullptr;
-      pShaderOp->VS = pShaderOp->GetString("VS66");
-      pShaderOp->PS = pShaderOp->GetString("PS66");
-      std::shared_ptr<ShaderOpTestResult> test = RunShaderOpTestAfterParse(
-          pDevice, m_support, "DynamicResourcesUniformIndexing", nullptr,
-          ShaderOpSet);
-
-      MappedData resultVSData;
-      MappedData resultPSData;
-      test->Test->GetReadBackData("g_resultVS", &resultVSData);
-      test->Test->GetReadBackData("g_resultPS", &resultPSData);
-      const float *resultVSFloats = (float *)resultVSData.data();
-      const float *resultPSFloats = (float *)resultPSData.data();
-      D3D12_QUERY_DATA_PIPELINE_STATISTICS Stats;
-      test->Test->GetPipelineStats(&Stats);
-
-
-      // VS
-      VERIFY_ARE_EQUAL(resultVSFloats[0], 10.0F);
-      VERIFY_ARE_EQUAL(resultVSFloats[1], 11.0F);
-      VERIFY_ARE_EQUAL(resultVSFloats[2], 12.0F);
-      VERIFY_ARE_EQUAL(resultVSFloats[3], 23.0F);
-      VERIFY_ARE_EQUAL(resultVSFloats[4], 24.0F);
-      VERIFY_ARE_EQUAL(resultVSFloats[5], 25.0F);
-      VERIFY_ARE_EQUAL(resultVSFloats[6], 30.0F);
-      VERIFY_ARE_EQUAL(resultVSFloats[7], 31.0F);
+      // Test Compute shader
+      {
+        pShaderOp->CS = pShaderOp->GetString("CS66");
+        std::shared_ptr<ShaderOpTestResult> test = RunShaderOpTestAfterParse(
+            pDevice, m_support, "DynamicResourcesDynamicIndexing", nullptr,
+            ShaderOpSet);
 
-      // PS
-      VERIFY_ARE_EQUAL(resultPSFloats[0], 10.0F);
-      VERIFY_ARE_EQUAL(resultPSFloats[1], 11.0F);
-      VERIFY_ARE_EQUAL(resultPSFloats[2], 12.0F);
-      VERIFY_ARE_EQUAL(resultPSFloats[3], 23.0F);
-      VERIFY_ARE_EQUAL(resultPSFloats[4], 24.0F);
-      VERIFY_ARE_EQUAL(resultPSFloats[5], 25.0F);
-      VERIFY_ARE_EQUAL(resultPSFloats[6], 30.0F);
-      VERIFY_ARE_EQUAL(resultPSFloats[7], 31.0F);
+        MappedData resultData;
+        test->Test->GetReadBackData("g_result", &resultData);
+        const float *resultCSFloats = (float *)resultData.data();
+
+        TestReadbackDynamicResourcesDynamicIndexing(non_uniform_bit, resultCSFloats, expectedResultsArray, expectedResultsSize);
```

Comment:
Could simply index `expectedResultsArray` here (or set a ref at the beginning of the loop), making it unnecessary to pass `non_uniform_bit` to the function, and for the function to even have to know anything about this.